app/src/test/kotlin/compiler/intermediate/ExpressionControlFlowTest.kt

```diff
@@ -673,4 +690,83 @@ class ExpressionControlFlowTest {
                 )
             )
     }
+
+    private fun Int.toLiteral() = Expression.NumberLiteral(this.toLong())
+    private fun Int.toConst() = IFTNode.Const(this.toLong())
+
+    @Test
+    fun `simple array`() {
```

Comment:
Some code snippets representing the tested operation would be nice (something similar to the tests above).